src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Write.Document.cs

```diff
@@ -105,39 +105,35 @@ public static JsonDocument SerializeToDocument(object? value, Type inputType, Js
                 ThrowHelper.ThrowArgumentNullException(nameof(context));
             }
 
-            Type runtimeType = GetRuntimeTypeAndValidateInputType(value, inputType);
-            return WriteDocumentUsingGeneratedSerializer(value, GetTypeInfo(context, runtimeType));
+            ValidateInputType(value, inputType);
+            return WriteDocumentAsObject(value, GetTypeInfo(context, inputType));
```

Comment:
I'm curious - could user possibly write type discriminators for `JsonTypeInfo<object>` and expect them to be deserialized directly rather than through JsonNode?

Reply:
Possibly -- although that would end up using the type system directly rather than a pure DOM. That being said, type discriminators on `object` are explicitly prohibited for the moment.